eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1686,6 +1690,10 @@ private long getLastSuccessfulHeartbeatTimePeriodInternal() {
     @com.netflix.servo.annotations.Monitor(name = METRIC_REGISTRY_PREFIX + "lastSuccessfulRegistryFetchTimePeriod",
             description = "How much time has passed from last successful local registry update", type = DataSourceType.GAUGE)
     private long getLastSuccessfulRegistryFetchTimePeriodInternal() {
+        if (!clientConfig.shouldRegisterWithEureka() || isShutdown.get()) {
```

Comment:
Not sure if I understand the first part of this condition, but maybe your intention was to check `shouldFetchRegistry` here and `shouldRegisterWithEureka` above?

Reply:
Yes, these two were reversed.